converter/shieldhit/geo.py

```diff
@@ -0,0 +1,191 @@
+from converter.solid_figures import SolidFigure, BoxFigure, CylinderFigure, SphereFigure
+from abc import ABC
+from dataclasses import dataclass, field
+from math import log10, ceil
+
+
+def format_float(number: float, n: int) -> float:
+    """
+    Format float to be up to n characters wide, as precise as possible and as short
+    as possible (in descending priority). so for example given 12.333 for n=5 you will
+    get 12.33, n=7 will be 12.333
+    """
+    # If number is zero we just want to get 0.0 (it would mess up the log10 operation below)
+    if number == 0:
+        return 0.0
+
+    lenght = n
+
+    # Adjust lenght for decimal separator ('.')
+    lenght -= 1
+
+    # Sign messes up the log10 we use do determine how long the number is. We use
+    # abs() to fix that, but we need to remember the sign and update `n` accordingly
+    sign = 1
+
+    if number < 0:
+        number = abs(number)
+        sign = -1
+        # Adjust lenght for the sign
+        lenght -= 1
+
+    whole_length = ceil(log10(number))
+
+    # Check if it will be possible to fit the number
+    if whole_length > lenght-1:
+        raise ValueError(f"Number is to big to be formatted. Minimum lenght: {whole_length-sign+1},\
+             requested length: {n}")
+
+    # Adjust n for the whole numbers, log returns resonable outputs for values greater
+    # than 1, for other values it returns nonpositive numbers, but we would like 1
+    # to be returned. We solve that by taking the greater value between the returned and
+    # and 1.
+    lenght -= max(whole_length, 1)
+
+    result = float(sign*round(number, lenght))
+
+    # Check if the round function truncated the number, warn the user if it did.
+    if result != sign*number:
```

Comment:
Float comparison is dangerous, in python `0.1 + 0.2` is not equal to `0.3`.
Better to use `math.isclose` with reasonable parameters